effects/gang_effect.py

```diff
@@ -0,0 +1,52 @@
+import message_text_config as msg
+import utils
+from citizens.citizen import Citizen
+from citizens.player import Player
+from game import Game
+
+from effects.effect import Effect, EffectStatus, InputStatusCode
+
+
+class GangEffect(Effect):
+    def __init__(self, context: 'Context', name: str,
+                 creator: Citizen) -> None:
+        super().__init__(context, name, creator, 13)
+
+    def _activate_impl(self) -> bool:
+        self.targets.append(self.creator)
+        self.user_interaction.save_active(msg.GangMessages.ACTIVATION_SUCCESS)
+
+        return True
+
+    def _resolve_impl(self) -> bool:
+        if self.game.round_number == self.activation_round:
+            return False
+
+        card_effects_list = []
+        citizens = self.context.game.citizens
+
+        for citizen in citizens:
+            card_effect = (effect for effect in citizen.effects
+                           if self.is_valid_card_effect(effect))
+
+            for effect in card_effect:
+                card_effects_list.append(effect)
+
+        for effect in card_effects_list:
+            effect.deactivate()
+            self.user_interaction.save_active(msg.GangMessages.RESOLVE_SUCCESS)
+
+        return True
+
+    def is_valid_card_effect(self, effect: 'Effect') -> bool:
```

Comment:
is_valid_card_effect.
Name is confusing too.
Seems like it relates somehow with validation of user input (_validate method)

Reply:
Don't use 'valid' word in this method name